numpy/testing/tests/test_utils.py

```diff
@@ -916,6 +916,17 @@ def test_timedelta(self):
         a = np.array([[1, 2, 3, "NaT"]], dtype="m8[ns]")
         assert_allclose(a, a)
 
+    def test_error_message_unsigned(self):
+        """Check the message is formatted correctly for the decimal value.
+           Also check the message when input includes inf or nan (gh12200)"""
+        x = np.asarray([0, 1, 2], dtype='uint8')
```

Comment:
```suggestion
        x = np.asarray([0, 1, 8], dtype='uint8')
```
would make the test check both ways :).

Reply:
Hmm. Ok. Let me adda comment.